modules/shredder/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/shredder/batch/Discovery.scala

```diff
@@ -90,6 +94,29 @@ object Discovery {
     (incomplete, unshredded)
   }
 
+  /**
+   * Most likely folder names should have "s3://path/run=YYYY-MM-dd-HH-mm-ss" format.
+   * This function extracts timestamp in the folder names and returns the ones in the
+   * given interval. Folders that does not have this format are included in the result also.
+   */
+  def findIntervalFolders(folders: List[Folder], since: Option[Instant], until: Option[Instant]): List[Folder] =
```

Comment:
I guess this function would be more readable if it would be possible to use string interpolation in pattern matching and `partitionMap`. But these are not usable on Scala 2.12.14. I hope it is also clear enough like this. Please suggest any change if you think it is not.

Reply:
The function looks fine, (I certainly can parse what's going on there) as long as it has a good test suite. And it does - tests look awesome. I'd just write one or two specifically for `findIntervalFolders` instead of testing it through `getState`